R/epiparameter_db.R

```diff
@@ -0,0 +1,765 @@
+  pathogen_ <- pathogen
+  epi_name_ <- epi_name
+
+  # clean input strings
+  disease <- .clean_string(disease)
+  pathogen <- .clean_string(pathogen)
+  epi_name <- .clean_string(epi_name)
+
+  # get valid options from db
+  disease_db <- vapply(
+    multi_epiparameter, function(x) x$disease,
+    FUN.VALUE = character(1)
+  )
+  pathogen_db <- vapply(
+    multi_epiparameter, function(x) x$pathogen,
+    FUN.VALUE = character(1)
+  )
+  epi_name_db <- vapply(
+    multi_epiparameter, function(x) x$epi_name,
+    FUN.VALUE = character(1)
+  )
+  disease_db <- c("all", .clean_string(unique(disease_db)))
+  pathogen_db <- c("all", .clean_string(unique(pathogen_db)))
+  epi_name_db <- c("all", .clean_string(unique(epi_name_db)))
+
+  # partial matching and custom error msg
+  tryCatch(
+    {
+      disease <- match.arg(
+        arg = disease,
+        choices = disease_db,
+        several.ok = FALSE
+      )
+      pathogen <- match.arg(
+        arg = pathogen,
+        choices = pathogen_db,
+        several.ok = FALSE
+      )
+      epi_name <- match.arg(
+        arg = epi_name,
+        choices = epi_name_db,
+        several.ok = FALSE
+      )
+    },
+    error = function(cnd) {
+      disease_str <- ifelse(
+        test = disease == "all", yes = "", no = paste(" for", disease_)
+      )
+      pathogen_str <- ifelse(
+        test = pathogen == "all", yes = "", no = paste(" for", pathogen_)
+      )
+      epi_name_str <- ifelse(test = epi_name == "all", yes = "", no = epi_name_)
+      stop(
+        epi_name_str, " distribution not available", disease_str, pathogen_str,
+        call. = FALSE
+      )
+    }
+  )
+
+  # filter based on disease
+  if (disease != "all") {
+    multi_epiparameter <- Filter(
+      f = function(x) .clean_string(x$disease) == disease,
+      x = multi_epiparameter
+    )
+  }
+
+  # filter based on pathogen
+  if (pathogen != "all") {
+    multi_epiparameter <- Filter(
+      f = function(x) .clean_string(x$pathogen) == pathogen,
+      x = multi_epiparameter
+    )
+  }
+
+  # filter by epi dist
+  if (epi_name != "all") {
+    multi_epiparameter <- Filter(
+      f = function(x) .clean_string(x$epi_name) == epi_name,
+      x = multi_epiparameter
+    )
+  }
+
+  # return multi_epiparameter
+  multi_epiparameter
+}
+
+#' Format data from JSON database into `<epiparameter>` objects
+#'
+#' @param x A single entry (element/object) from the database.
+#'
+#' @return An `<epiparameter>` object.
+#' @keywords internal
+#' @noRd
+.format_epiparameter <- function(x) {
+  # convert nulls to NA
+  null2na <- function(l) {
+    if (is.list(l)) l <- lapply(l, null2na) else if (is.null(l)) l <- NA
+    l
+  }
+  x <- lapply(x, null2na)
+
+  # format parameters and parameter uncertainty
+  params_uncertainty <- .format_params(x)
+  params <- params_uncertainty$params
+  uncertainty <- params_uncertainty$uncertainty
+
+  # TODO: remove this and move precision from parameter to summary statistic
```

Comment:
Is this todo dependent on something external or should this instead be moved to a GitHub issue to track? 

These kinds of comments are easy to lose track of so I would encourage to put it in an issue, fix it, or remove it altogether.

Reply:
Thanks, I've logged in #400 and will resolve in the next version and will remove this `# TODO` comment.